### PR TITLE
Rewrite all teams page into React.js

### PR DESCRIPTION
**Pull request by ZmagoD**
https://biosistemika.atlassian.net/browse/SCI-1497

### CODE REVIEW

File: app/controllers/client_api/users/user_teams_controller.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  module Users
+    class UserTeamsController < ApplicationController
+      include NotificationsHelper
+      include InputSanitizeHelper
+      include ClientApi::Users::UserTeamsHelper
+
+      before_action :find_user_team, only: :leave_team
+
+      def leave_team
+        if user_cant_leave
```
Comment (programmertoni) [inline app/controllers/client_api/users/user_teams_controller.rb]: Predicate functions that return Boolean usually have '?' at the end of function name

File: app/javascript/packs/shared/modals_container/modals/LeaveTeamModal.jsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  }
+
+  leaveTeam() {
+    const teamUrl = `${LEAVE_TEAM_PATH}?team=${this.props.teamId}`;
+    axios
+      .delete(teamUrl, {
+        withCredentials: true
+      })
+      .then(response => {
+        console.log(response);
```
Comment (programmertoni) [inline app/javascript/packs/shared/modals_container/modals/LeaveTeamModal.jsx]: you can delete this console.log

File: app/javascript/packs/shared/modals_container/modals/LeaveTeamModal.jsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        const teams = response.data.teams.collection;
+        this.props.addTeamsData(teams);
+        const currentTeam = _.find(teams, team => team.current_team);
```
Comment (programmertoni) [inline app/javascript/packs/shared/modals_container/modals/LeaveTeamModal.jsx]: I think you need braces around _.find((teams, team) => ...

only if you have one variable in fat arrow function you can leave braces off

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -149,4 +149,44 @@
...
+
+  describe 'teams_data should return a list of teams' do
+    #  needs persistence because is testing a sql query
+    let(:team) { create :team }
+    let(:user_one) do
+      create :user, email: 'user1@asdf.com', current_team_id: team.id
+    end
+    let(:user_two) { create :user, email: 'user2@asdf.com' }
+
+    it 'in a specific format' do
```
Comment (programmertoni) [inline spec/models/user_spec.rb]: Maybe you can write this example to be more readable.

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -149,4 +149,44 @@
...
+        members: 2,
+        role: 2,
+        current_team: true
+      }
+
+      user_one.teams_data.first.each do |k, v|
+        expect(v).to eq(expected_result.fetch(k.to_sym))
+      end
+    end
+  end
```
Comment (programmertoni) [inline spec/models/user_spec.rb]: There should be one more test case

3 teams (or 2) and one user does not belong to this team

File: app/models/user.rb [hunk 6]
```diff
@@ -204,6 +204,24 @@ def current_team
     Team.find_by_id(self.current_team_id)
   end
 
+  # Retrieves the data needed in all teams page
+  def teams_data
+    ActiveRecord::Base.connection.execute(
+      ActiveRecord::Base.send(
```
[the remark this answers is not in the document]
Comment (ZmagoD) [reply app/models/user.rb]: is a private method that is invoked you s...

File: app/javascript/packs/shared/modals_container/modals/LeaveTeamModal.jsx (hunk 3 above)
Comment (ZmagoD) [reply app/javascript/packs/shared/modals_container/modals/LeaveTeamModal.jsx]: first arg is array the second is function

Comment (programmertoni) [reply app/javascript/packs/shared/modals_container/modals/LeaveTeamModal.jsx]: :+1:

**Pull request merged**